src/core/tactics.ml

```diff
@@ -8,65 +8,127 @@ open Pos
 open Syntax
 open Proof
 open Print
+open Timed
 
 (** Logging function for tactics. *)
 let log_tact = new_logger 't' "tact" "tactics"
 let log_tact = log_tact.logger
 
+(** [solve ps pos] calls the default solve algorithm on the unification
+    goals of the proof state [ps] and does not fail *)
```

Comment:
You say that this function does not fail but it calls fatal when constraints are unsolvable. Do you want to say that it fails only if some constraint is unsolvable?

Reply:
yes
fixed